DOCS-9998-sftp-index-studio-editorial-k (#1258)
mulesoft/docs-connectors: sftp/1.3/modules/ROOT/edits/sftp-studio.adoc

[[create-mule-project]]
== Create a Mule Project

In Studio, create a new Mule project in which to add and configure the connector:

. In Studio, select *File > New > Mule Project*.
. Enter a name for your Mule project and click *Finish*.

[[add-connector-to-project]]
== Add the Connector to Your Mule Project

Add SFTP Connector to your Mule project to automatically populate the XML code with the connector's namespace and schema location and to add the required dependencies to the project's `pom.xml` file:

. In the *Mule Palette* view, click *(X) Search in Exchange*.
. In the *Add Dependencies to Project* window, type `sftp` in the search field.
. Click *SFTP Connector* in *Available modules*.
. Click *Add*.
. Click *Finish*.

Adding a connector to a Mule project in Studio does not make that connector available to other projects in your Studio workspace.

[[configure-input-source]]
== Configure a Source

A source initiates a flow when a specified condition is met.
You can configure one of these sources to use with SFTP Connector:

* On New or Updated File +

> Initiates a flow by polling a directory for files that have been created or updated. A message is generated for each file that is found.

Reviewer: attempt to make one sentence, active voice
```suggestion
Initiates a flow by polling a directory for files that have been created or updated and generates a message for each file found.
```

* HTTP Listener +

> Initiates a flow each time it receives a request on the configured host and port

Reviewer: ```suggestion
Initiates a flow each time it receives a request on the configured host and port.
```

* Scheduler +

> Initiates a flow when a time-based condition is met

Reviewer: ```suggestion
Initiates a flow when a time-based condition is met.
```

For example, to configure the On New or Updated File source, follow these steps:

. In the *Mule Palette* view, select *On New or Updated File*.
. Drag *On New or Updated File* to the Studio canvas.
. On the *On New or Updated File* configuration screen, optionally change the value of the *Display Name* field.
. Specify a value for the *Path* field.
. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of *On New or Updated File* in the app.
. On the *General* tab, specify the connection information for the connector.

> . Set the *Host* field to specify the SFTP server host, for example `localhost`, `192.168.0.1`, and so on.

Reviewer: ```suggestion
. Set the *Host* field to specify the SFTP server host, for example, `localhost` or `192.168.0.1`.
```

. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

> . On the *On New or Updated File* configuration screen, specify a *Scheduling Strategy*, and other optional fields.

Reviewer: ```suggestion
. On the *On New or Updated File* configuration screen, specify a *Scheduling Strategy* value and optionally complete other fields.
```

To add an operation for SFTP Connector, follow these steps:

. In the *Mule Palette* view, select *SFTP* and then select the desired operation.
. Drag the operation onto the Studio canvas and to the right of the source.

The following screenshot shows the SFTP Connector operations in the Mule Palette view of Anypoint Studio:

.SFTP Connector Operations
image::sftp-operations.png[SFTP Connector Operations]

[[configure-global-element]]
== Configure a Global Element for the Connector

When you configure a connector, it’s best to configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to configure the SFTP server connection.

To configure the global element for SFTP Connector, follow these steps:

. Select the name of the connector in the Studio canvas.
. Select the operation in the Studio canvas.
. In the configuration screen for the operation, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
. On the *General* tab, specify the connection information for the connector:

> . Set the *Working Directory* as the root of every relative path used with this connector. If it's not provided, the value defaults to the remote server.

Reviewer: ```suggestion
. Set the *Working Directory* as the root of every relative path used with this connector. 
+
If it's not provided, the value defaults to the remote server.
```

> . Set the *Host* field to specify the SFTP server host, for example `localhost` or `192.168.0.1`.

Reviewer: ```suggestion
. Set the *Host* field to specify the SFTP server host, for example, `localhost` or `192.168.0.1`.
```

> . On the *Advanced* tab, optionally specify timeout configuration and reconnection strategy.

Reviewer: ```suggestion
. On the *Advanced* tab, optionally specify a timeout configuration and reconnection strategy.
```
